src/01_getting_started/03_state_of_async_rust.md

```diff
@@ -1,24 +1,91 @@
-afraid of doing some spelunking, enjoy your dive into the world of
-asynchronous programming in Rust!
+Parts of async Rust are supported with the same stability guarantees as
+regular Rust. Other parts are still maturing and will change over time.
+Compared to regular Rust, you can expect:
 
+- Outstanding runtime performance.
+- More frequent interaction with advanced language features, such as lifetimes
+  and pinning.
+- Some compatibility constraints, both between sync and async code, and between
+  different async ecosystems.
+- Higher maintenance burden, due to the ongoing evolution of async ecosystems
+  and language support.
+
+In short, async Rust is more difficult to use and can result in a higher
+maintenance burden, but gives you best-in-class performance in return.
+All areas of async Rust are constantly improving,
+so the impact of these issues will wear off over time.
+
+## Language and library support
+
+While asynchronous programming is supported by Rust itself,
+most async applications depend on functionality provided
+by community crates.
+As such, you need to rely on a mixture of
+language features and library support:
+
+- The most fundamental traits, types and functions, such as the
+  [`Future`](https://doc.rust-lang.org/std/future/trait.Future.html) trait
+  are provided by the standard library.
+- The `async/await` syntax is supported directly by the Rust compiler.
+- Many utility types, macros and functions are provided by the
+  [`futures`](https://docs.rs/futures/) crate. They can be used in any async
+  Rust application.
+- Async runtimes, IO and task spawning are provided by "async ecosystems", such
+  as Tokio and async-std. Most async applications, and some async crates,
+  depend on a specific ecosystem. See
+  ["The Async Ecosystem"](../02_execution/01_chapter.md) section for more
+  details.
+
+Some language features you may be used to from regular Rust are not yet
+available in async Rust. Notably, Rust does not let you declare async
+functions in traits. Instead, you need to use workarounds to achieve the same
+result, which can be more verbose.
+
+## Compiling and debugging
+
+Compilation errors in async Rust conform to the same high standards as
+regular Rust, but since async Rust often depends on more complex language
+features, such as lifetimes and pinning, you may encounter these types of
+errors more frequently.
+
+Runtime errors, or panics, also tend to be more verbose, primarily as a
+result of the state machines that asynchronous functions compile to. Stack
+traces in async Rust also contain functions from the async runtime you are
+using.
+
+Lastly, there are a couple of new failure modes in async Rust, for instance
+if you call a blocking function from an async context or if you implement
+the `Future` trait incorrectly. Such errors can silently pass both the
+compiler and even unit tests. At the moment, you need
+to be aware of these issues and learn how to avoid them.
```

Comment:
Can you link to somewhere with more info on these issues? Or maybe just give a bit more detail on what can go wrong? I wouldn't say "you need to be aware of these issues and how to avoid them" without at least linking to some more context.